### ISSUE TITLE
`rename` should support hash-qualified names for the source name (but not for the target name)

### ISSUE THREAD

**Issue by pchiusano** [unisonweb/unison#313]
That is, should be able to write:

    > rename foo#9sdjf oldFoo

**Comment by aryairani** [comment]
Still relevant in #354

**Comment by pchiusano** [comment]
Is this done? I thought so... if yes you can close. @aryairani

**Comment by runarorama** [comment]
Is this still relevant?

**Comment by pchiusano** [comment]
```
> rename.term foo.bar#asldfkj

  ⚠️
  
  `rename.term` takes two arguments, like `rename.term oldname newname`.

> rename.term foo.bar#34 baz

  extra characters after foo.bar: "#34"
```

**Comment by aryairani** [comment]
@runarorama We do have a similar problem in branchless, though maybe for a different reason now.

I'd start by looking at how `Path.parseHQ'Split'` behaves on `foo.bar#34`.  Does it return `Right` or `Left`?

**Comment by aryairani** [comment]
Fixed in bfbec5a
![image](https://user-images.githubusercontent.com/538571/60341394-530f0b00-997c-11e9-96f8-0655e5f7c5d9.png)
